web/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -16,31 +20,57 @@ export class DashboardComponent implements OnInit {
 
   constructor(
     private userService: UserService,
+    private route: ActivatedRoute,
+    private errorService: ErrorService
   ) { }
 
   ngOnInit() {
     this.isResponseLoading = true;
+    this.route.snapshot.paramMap.keys.length ?
+      this.displayPublicDashboard() :
+      this.displayOwnDashboard();
+  }
+
+  displayOwnDashboard(): void {
     this.userService.getUser()
-      .subscribe((user) => {
-        if (!user) {
-          console.error('error in fetching user'); // TODO: Handle null user scenario in dashboard UI
+      .pipe(
+        tap(user => {
+          this.user = { ...user };
```

Comment:
This can be a single line to be consistent but don't worry about changing it now, it can be done another time